packages/client/src/monaco-vscode-api-services.ts

```diff
@@ -40,50 +32,112 @@ export type InitializeServiceConfig = {
+            const {
+                default: getModelEditorServiceOverride
+            } = await import('vscode/service-override/modelEditor');
+            const defaultOpenEditorFunc: OpenEditor = async (model, options, sideBySide) => {
+                console.log('Trying to open a model', model, options, sideBySide);
+                return undefined;
+            };
+            if (config.modelEditorServiceConfig.useDefaultFunction) {
+                allServices.modelService = getModelEditorServiceOverride(defaultOpenEditorFunc);
+            } else if (config.modelEditorServiceConfig.openEditorFunc) {
+                allServices.modelService = getModelEditorServiceOverride(config.modelEditorServiceConfig.openEditorFunc);
+            }
+        }
+
+        if (config.enableConfigurationService === true && config.configurationServiceConfig) {
+            const { default: getConfigurationServiceOverride } = await import('vscode/service-override/configuration');
+            allServices.configurationService = getConfigurationServiceOverride(config.configurationServiceConfig.defaultWorkspaceUri);
+        }
+
+        if (config.enableThemeService === true) {
+            const { default: getThemeServiceOverride } = await import('vscode/service-override/theme');
+            allServices.themeService = getThemeServiceOverride();
+
+            // theme requires textmate
+            config.enableTextmateService = true;
+        }
+
+        if (config.enableKeybindingsService === true) {
+            const { default: getKeybindingsServiceOverride } = await import('vscode/service-override/keybindings');
+            allServices.keybindingsService = getKeybindingsServiceOverride();
+        }
+
+        if (config.enableTextmateService === true) {
+            const { default: getTextmateServiceOverride } = await import('vscode/service-override/textmate');
+            allServices.textmateService = getTextmateServiceOverride();
+        }
+
+        if (config.enableLanguagesService === true) {
+            const { default: getLanguagesServiceOverride } = await import('vscode/service-override/languages');
+            allServices.languagesService = getLanguagesServiceOverride();
+        }
+
+        if (config.enableAudioCueService === true) {
+            const { default: getAudioCueServiceOverride } = await import('vscode/service-override/audioCue');
+            allServices.audioCueService = getAudioCueServiceOverride();
+        }
+
+        if (config.enableDebugService === true) {
+            const { default: getDebugServiceOverride } = await import('vscode/service-override/debug');
+            allServices.debugService = getDebugServiceOverride();
+        }
+
+        if (config.enablePreferencesService === true) {
+            const { default: getPreferencesServiceOverride } = await import('vscode/service-override/preferences');
+            allServices.preferencesService = getPreferencesServiceOverride();
+        }
```

Comment:
It would probably be a good idea to load them all at once with a Promise.all()

also it would be simpler to just return a single `ServiceOverride`

Reply:
Yes, true. I will look into that.